### PR TITLE
Same person stop registering again

### PR DESCRIPTION
**Pull request by Pilkas-git**

### CODE REVIEW

File: app-android-xamarin/Moodis/Moodis/Feature/SignIn/SignInActivity.cs [hunk 1]
```diff
@@ -190,5 +197,18 @@ private void SignIn(string username, string password)
                 Toast.MakeText(this, Resource.String.user_not_found_error, ToastLength.Short).Show();
             }
         }
+
+        private void DisplayFaceUpdatewWindow()
+        {
+            Android.Support.V7.App.AlertDialog.Builder builder = new Android.Support.V7.App.AlertDialog.Builder(this)
```
Comment (tmaciulis22) [inline app-android-xamarin/Moodis/Moodis/Feature/SignIn/SignInActivity.cs]: Use an AppCompatExtension here

File: app-android-xamarin/Moodis/Moodis/Network/Face/Face.cs [hunk 2]
```diff
@@ -135,10 +132,37 @@ public async Task<IList<Person>> IdentifyPersons(string imageFilePath, Action<Li
                     }
                 }
             }
-
             return personsList;
         }
 
+        //returns true if one or more users have accounts false otherwise.
+        public async Task<Boolean> MultipleAccounts(string imageFilePath, Action<List<DetectedFace>> callback)
```
Comment (tmaciulis22) [inline app-android-xamarin/Moodis/Moodis/Network/Face/Face.cs:139]: Wouldn't it be just enough to check if the indentified people list is empty or not? Like in the image below
image

File: app-android-xamarin/Moodis/Moodis/Resources/values/strings.xml [hunk 3]
```diff
@@ -37,6 +37,7 @@
   <string name="stronger_password">Password must be stronger</string>
   <string name="user_created">User created successfully</string>
   <string name="user_exists">User already exists</string>
+  <string name="user_face_exists">A user with such a face already exists try to sign in.</string>
```
Comment (tmaciulis22) [inline app-android-xamarin/Moodis/Moodis/Resources/values/strings.xml]: Just a suggestion, but maybe this string should be: `You already have an account registered in Moodis`

File: app-android-xamarin/Moodis/Moodis/Resources/values/strings.xml [hunk 4]
```diff
@@ -49,6 +50,9 @@
 
   <string name="logout">Logout</string>
   <string name="logout_confirmation_message">Are you sure you want to sign out ?</string>
+
+  <string name="update_face">Update Profile</string>
+  <string name="update_face_confirmation_message">Would you like to update your face profile ? (You will be able to sign in with face next time you log in.)</string>
```
Comment (tmaciulis22) [inline app-android-xamarin/Moodis/Moodis/Resources/values/strings.xml]: remove space before `?`

File: app-android-xamarin/Moodis/Moodis/Feature/SignIn/SignInActivity.cs [hunk 5]
```diff
@@ -190,5 +197,18 @@ private void SignIn(string username, string password)
                 Toast.MakeText(this, Resource.String.user_not_found_error, ToastLength.Short).Show();
             }
         }
+
+        private void DisplayFaceUpdatewWindow()
```
Comment (tmaciulis22) [inline app-android-xamarin/Moodis/Moodis/Feature/SignIn/SignInActivity.cs]: typo: `wWindow`

File: app-android-xamarin/Moodis/Moodis/Feature/SignIn/SignInActivity.cs [hunk 6]
```diff
@@ -53,6 +54,11 @@ protected override void OnActivityResult(int requestCode, [GeneratedEnum] Result
                 SetResult(Result.Ok, new Intent().PutExtra(EXTRA_SIGNED_IN, true));
                 Finish();
             }
+            else if (resultCode == Result.Ok && requestCode == REQUEST_CODE_REGISTER_FACE)
```
Comment (tmaciulis22) [inline app-android-xamarin/Moodis/Moodis/Feature/SignIn/SignInActivity.cs]: Why was this added?

File: app-android-xamarin/Moodis/Moodis/Feature/Register/RegisterFaceActivity.cs [hunk 7]
```diff
@@ -190,6 +200,21 @@ private void InitEventHandler()
             };
         }
 
+        private async System.Threading.Tasks.Task CheckWhetherUserFaceAlreadyUsedAsync(string imagePath)
```
Comment (tmaciulis22) [inline app-android-xamarin/Moodis/Moodis/Feature/Register/RegisterFaceActivity.cs]: Shorten return type

File: app-android-xamarin/Moodis/Moodis/Feature/Register/RegisterFaceActivity.cs [hunk 8]
```diff
@@ -41,6 +44,7 @@ protected override void OnCreate(Bundle savedInstanceState)
 
             PhotosLeft = FindViewById<TextView>(Resource.Id.photosLeft);
             PhotosLeft.Text = GetString(Resource.String.register_face_photos_left, RegisterViewModel.RequiredNumberOfPhotos - registerViewModel.photosTaken);
+            updating = Intent.GetBooleanExtra("update", false);
```
Comment (tmaciulis22) [inline app-android-xamarin/Moodis/Moodis/Feature/Register/RegisterFaceActivity.cs]: "update" should be moved to const string `EXTRA_UPDATE`

**Review by tmaciulis22 — CHANGES_REQUESTED** [review]
🔍

File: app-android-xamarin/Moodis/Moodis/Network/Face/Face.cs (hunk 2 above)
Comment (Pilkas-git) [reply app-android-xamarin/Moodis/Moodis/Network/Face/Face.cs:139]: Yeah that would be a lot better. Can't believe i didn't see it earlier.

File: app-android-xamarin/Moodis/Moodis/Feature/SignIn/SignInActivity.cs (hunk 6 above)
Comment (Pilkas-git) [reply app-android-xamarin/Moodis/Moodis/Feature/SignIn/SignInActivity.cs]: If user signs in normaly he is prompt to update face. After face is updated he is logged in like normal..

**Comment by Pilkas-git** [comment]
Added personId to local database as the api one is not yet merged. Also why were we not saving it and just ignoring it in the first place ? :DD

File: app-android-xamarin/Moodis/Moodis/Feature/Register/RegisterViewModel.cs [hunk 9]
```diff
@@ -52,8 +55,14 @@ public async Task<Response> DeleteUser()
             return await Face.Instance.DeletePerson(currentUser.PersonGroupId);
         }
 
+        //TODO MOVE THIS TO FACE CLASS
         public async Task<Response> AddFaceToPerson(string imagePath)
         {
+            if (currentUser == null)
+            {
+                currentUser = SignInViewModel.currentUser;
```
Comment (tmaciulis22) [inline app-android-xamarin/Moodis/Moodis/Feature/Register/RegisterViewModel.cs]: why do you need this assignment?

File: app-android-xamarin/Moodis/Moodis/Network/Face/Face.cs [hunk 10]
```diff
@@ -166,11 +177,9 @@ public async Task<Response> AddFaceToPerson(string imageFilePath, string personG
         {
             try
             {
-                imageFilePath.RotateImage();
-
                 using Stream imageFileStream = File.OpenRead(imageFilePath);
                 await faceClient.PersonGroupPerson.AddFaceFromStreamAsync(personGroupId,
-                    user.FaceApiPerson.PersonId, imageFileStream);
+                    Guid.Parse(user.personId), imageFileStream); ;
```
Comment (tmaciulis22) [inline app-android-xamarin/Moodis/Moodis/Network/Face/Face.cs]: two semicolons

**Review by tmaciulis22 — CHANGES_REQUESTED** [review]
Minor changes

File: app-android-xamarin/Moodis/Moodis/Feature/Register/RegisterViewModel.cs (hunk 9 above)
Comment (Pilkas-git) [reply app-android-xamarin/Moodis/Moodis/Feature/Register/RegisterViewModel.cs]: it tries to take current user from register view model in which currentUser is not defined because the person is not registering, but isntead updating his face data.

Comment (tmaciulis22) [reply app-android-xamarin/Moodis/Moodis/Feature/Register/RegisterViewModel.cs]: But we use `RegisterViewModel's` `currentUser` in only few places? If so, then it might be better to call `SignInViewModel's` `currentUser`? I mean, it is much better to have only one source of truth for this property

Comment (Pilkas-git) [reply app-android-xamarin/Moodis/Moodis/Feature/Register/RegisterViewModel.cs]: Yeah we SHOULD have current user in only one place however as it is now it's in singinViewModel and RegisterViewModel which both get current user on diferent instances (registering/signin in).  Don't know why aurimas created a new one for registration purposes.

**Review by tmaciulis22 — APPROVED** [review]
Good job 💪

**Review by AArlauskas — APPROVED** [review]
